src/cmd/ksh93/sh/init.c

```diff
@@ -2146,8 +2146,6 @@ static void env_init(Shell_t *shp) {
             nv_newattr(np, flag | NV_IMPORT | NV_EXPORT, size);
             if ((flag & (NV_INTEGER | NV_UTOL | NV_LTOU)) == (NV_UTOL | NV_LTOU))
                 nv_mapchar(np, (flag & NV_UTOL) ? e_tolower : e_toupper);
-        } else {
-            cp += 2;
```

Comment:
Please also change the statement 11 lines below to hoist the initial assignment to its own statement:

```
    if ((cp = nv_getval(SHELLNOD)) && (sh_type(cp) & SH_TYPE_RESTRICTED)) {
```